pynobo.py

```diff
@@ -506,35 +553,38 @@ def response_handler(self, r):
             self.logger.warning('behavior undefined for this response: {}'.format(r))
             warnings.warn('behavior undefined for this response: {}'.format(r)) #overkill?
 
-    def create_override(self, mode, type, target_type, target_id='-1', end_time='-1', start_time='-1'):
-        """Override hub/zones/components. Use OVERRIDE_MODE_NOMAL to disable an existing override.  
-
-        Keyword arguments:  
-        mode -- API.OVERRIDE_MODE. NORMAL, COMFORT, ECO or AWAY
-        type -- API.OVERRIDE_TYPE. NOW, TIMER, FROM_TO or CONSTANT  
-        target_type -- API.OVERRIDE_TARGET. GLOBAL or ZONE  
-        target_id -- the target id (default -1)  
-        end_time -- the end time (default -1)  
-        start_time -- the start time (default -1)
-        """        
+        for callback in self.callbacks:
```

Comment:
I think we should move this. to `socket_receive`.

Currently callbacks are getting called for every message received, and since we do not tell the callback function what has changed, they might do an expensive update of everything every time they get called.

Reply:
I see your point, but then the initial state on a connect or reconnect would not be passed on. Adding a callback in `connect_hub` as well solves this, and also ensures that the callback is not sent until the complete initial state is received, avoiding callbacks for partial state - I'll look into that!

An expensive callback function is a violation of the contract:
```
        Register a callback to notify updates to the hub state. The callback MUST be safe to call
        from the event loop. The nobo instance is passed to the callback function.
```
I don't know how to enforce the contract though. Home Assistant uses a custom annotation `@homeassistant.core.callback`, but I don't know how HA would handle an unannotated callback.

Reply:
Callback moved to `socket_receive`.